### ISSUE TITLE
ML Foundation Model Uses Global Mutable State Shared Across Requests

### ISSUE THREAD

**Issue by ionfwsrijan** [KanishJebaMathewM/Truxify#11233]
# Issue: ML Foundation Model Uses Global Mutable State Shared Across Requests

## Description

The foundation model routes (`backend/ml/routes/foundation_routes.py`) initialize model, trainer, and processor as **global module-level variables** (lines 16-28):

```python
# Initialize model and config
config = FoundationModelConfig()
model = LogisticsFoundationModel(...)
trainer = FoundationModelTrainer(model, config)
processor = LogisticsDataProcessor()
```

**Critical problems**:

1. **Shared model weights across requests**: Concurrent `/predict` requests share the SAME model instance. PyTorch models are not thread-safe for concurrent forward passes with mutating state (dropout, batch norm).

2. **Trainer config mutation** (line 144-145):
   ```python
   config.epochs = epochs
   trainer.config = config
   ```
   - Modifies GLOBAL config
   - Affects ALL subsequent requests
   - Race condition: Request A sets epochs=5, Request B sets epochs=10, Request A trains with 10 epochs

3. **Processor vocab mutation**: `processor.save_vocab()` / `load_vocab()` (lines 242, 260) modify global vocab state

4. **No request isolation**: Training (`/pretrain`, `/finetune`) modifies global model weights while prediction requests may be running

5. **Model save/load race**: `/save` and `/load` operate on global model - concurrent save/load corrupts state

## Expected Behavior

- Each request gets isolated model/trainer/processor OR
- Proper locking for shared resources
- Training runs in separate process/worker
- Request-scoped config

## Actual Behavior

- Concurrent predictions may crash or return garbage
- Training config leaks across requests
- Vocab corruption under concurrent save/load
- No horizontal scaling possible (global state)

## Proposed Fix

1. **Request-scoped dependencies**: Use FastAPI `Depends` to create per-request instances
   ```python
   def get_model():
       return LogisticsFoundationModel(...)
   
   @router.post(/predict)
   async def predict(text: str, task: str = 'classification', model=Depends(get_model)):
   ```

2. **Separate training worker**: Move `/pretrain`, `/finetune` to background job queue (Celery/RQ) with separate process

3. **Model versioning**: Atomic model files with version, load specific version for prediction

4. **Read-only model serving**: Freeze model weights for prediction, only training worker modifies

5. **Add locking** if global state unavoidable:
   ```python
   import asyncio
   _model_lock = asyncio.Lock()
   
   @router.post(/predict)
   async def predict(...):
       async with _model_lock:
           # inference
   ```

6. **Remove config mutation**: Pass epochs as parameter to `trainer.train()` not via config

## Files to Modify

- `backend/ml/routes/foundation_routes.py` - Complete rewrite of dependency management
- `backend/ml/foundation/model.py` - Make trainer.train accept epochs param
- `backend/ml/foundation/data.py` - Ensure processor is stateless or request-scoped
- `backend/ml/requirements.txt` - Add Celery/RQ if not present
- `k8s/*` - Separate deployment for training worker

## Severity

**Critical** - Data corruption, incorrect predictions, training interference, cannot scale
